packages/core/src/types.ts

```diff
@@ -1692,22 +1687,12 @@ export declare namespace SCXML {
 }
 
 // Taken from RxJS
```

Comment:
This comment is not quite applicable right now - if anything it could mention "inspired by RxJS". They intentionally had this as a union. It looks a little bit dumb but prevents the consumer from passing an empty object. I'm not sure if it's important though - I'm OK with a simpler version.